lexers/nix.lua

```diff
@@ -0,0 +1,101 @@
+-- Nix LPEG lexer.
+-- Written by Samuel Marquis.
+
+local lexer = lexer
+local P, S = lpeg.P, lpeg.S
+
+local lex = lexer.new(...)
+
+-- Keywords.
+lex:add_rule('keyword', lex:tag(lexer.KEYWORD, lex:word_match(lexer.KEYWORD)))
+
+-- Functions.
+lex:add_rule('function', lex:tag(lexer.FUNCTION_BUILTIN, lex:word_match(lexer.FUNCTION_BUILTIN)))
```

Comment:
I just thought of this now, but would it make sense to only highlight functions if a '(' follows? This is the case for many lexers.

Reply:
Gonna have to double-check and make sure there's no Haskell-style calling convention weirdness, but that's probably not a bad idea

Reply:
[Yeah doesn't look like this works, it's Haskell-style.](https://nix.dev/manual/nix/2.18/language/builtins)